src/tools/PICS-generator/PICSGenerator.py

```diff
@@ -0,0 +1,429 @@
+import argparse
+import json
+import os
+import pathlib
+import sys
+import xml.etree.ElementTree as ET
+
+import chip.clusters as Clusters
+from rich.console import Console
+
+# Add the path to python_testing folder, in order to be able to import from matter_testing_support
+sys.path.append(os.path.abspath(sys.path[0] + "/../../python_testing"))
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main  # noqa: E402
+
+console = None
+
+
+def GenerateDevicePicsXmlFiles(clusterName, clusterPicsCode, featurePicsList, attributePicsList, acceptedCommandPicsList, generatedCommandPicsList, outputPathStr):
+
+    xmlPath = xmlTemplatePathStr
+    fileName = ""
+
+    print(f"Handling PICS for {clusterName}")
+
+    # Map clusters to common XML template if needed
+    otaProviderCluster = "OTA Software Update Provider Cluster"
+    otaRequestorCluster = "OTA Software Update Requestor Cluster"
+    onOffCluster = "On/Off Cluster"
+    groupKeyManagementCluster = "Group Key Management Cluster"
+    nodeOperationalCredentialsCluster = "Node Operational Credentials Cluster"
+    basicInformationCluster = "Basic Information Cluster"
+    networkCommissioningCluster = "Network Commissioning Cluster"
+
+    if otaProviderCluster in clusterName or otaRequestorCluster in clusterName:
+        clusterName = "OTA Software Update"
+
+    elif onOffCluster == clusterName:
+        clusterName = clusterName.replace("/", "-")
+
+    elif groupKeyManagementCluster == clusterName:
+        clusterName = "Group Communication"
+
+    elif nodeOperationalCredentialsCluster == clusterName or basicInformationCluster == clusterName or networkCommissioningCluster == clusterName:
```

Comment:
Not for this PR, but would it make life easier in the long run to standardize on filenames in the spec?

Reply:
Yes! Very much. I have not started poking at this yet but I want to get the ball rolling on aligning filenames across spec, SDK and test plan at some point. Hopefully soon.

Reply:
I have added this as a separate task on my backlog as well as the backlog for this tool to push for this 😄 I will resolve this for now.